lib/src/algorithms.dart

```diff
@@ -32,10 +32,10 @@ int _comparableBinarySearch/*<T extends Comparable<T>>*/(
 /// the objects.
 ///
 /// Returns -1 if [value] is not in the list by default.
-int binarySearch/*<T extends Comparable<T>>*/(
+int binarySearch/*<T>*/(
     List/*<T>*/ sortedList, /*=T*/ value, { int compare(/*=T*/ a, /*=T*/ b) }) {
   if (compare == null) {
-    return _comparableBinarySearch(sortedList, value);
+    return _comparableBinarySearch(sortedList as List<Comparable>, value as Comparable);
```

Comment:
> Also, we should more clearly document that this will throw a runtime error if compare() is omitted and T isn't a subtype of Comparable<T>.

Why isn't that sufficient?  Are you worried that the List will fail `as List<Comparable>` because it might be a `List<Object>` but that it will just-so-happen that all the elements of the list will actually pass `as Comparable` ?